### ISSUE TITLE
Support HttpRequestWithBody#body(InputStream) method

### ISSUE THREAD

**Issue by dzikoysk** [Kong/unirest-java#411]
**Is your feature request related to a problem? Please describe.**

I'd like to send large files, especially in tests, to check if server is able to large handle incoming data, e.g. with:

```groovy
RandomAccessFile hugeFile = new RandomAccessFile(new File(temptDirectory, "test.binary"), "rw")
hugeFile.setLength(1024 * 1024 * 1024) // 1GB
new InputStreamEntity(Channels.newInputStream(hugeFile.getChannel())) // :(
```

**Describe the solution you'd like**

Just a simple `HttpRequestWithBody#body(InputStream)` would be great. It was even referenced in the past as a possible feature, but looks like no one finally did this:

* https://github.com/Kong/unirest-java/issues/248#issuecomment-441503513

It shouldn't be that hard, as InputStream is already implemented in internals as one of available BodyPart, just not exposed as an public API.

**Describe alternatives you've considered**

Just using another http client for such a case, as literally every client I know support such an option, e.g. Apache's HttpClient:

```groovy
def put = new HttpPut(url("/large.binary").toString())
put.setEntity(new InputStreamEntity(Channels.newInputStream(hugeFile.getChannel())))
def response = client.execute(put)
```

**Comment by ryber** [comment]
This IS actually supported (sort of) as a named param file https://github.com/Kong/unirest-java/blob/main/unirest/src/test/java/BehaviorTests/MultiPartFormPostingTest.java#L64

It's true there is not a method as a raw body octet stream (not a named file)

**Comment by dzikoysk** [comment]
Yeah, but parametrized files are a part of forms to upload multiple files, it's slightly different case than just a huge body of the request itself I guess 🤔

**Comment by ryber** [comment]
this is released in 3.13.0. Try it out and let me know what you think. You can also add a progress monitor to it (though it doesn't know the total size of the stream of course, so you just get the bits written and the total of bytes written on each chunk.

**Comment by dzikoysk** [comment]
Works like a charm for me now :)

* https://github.com/dzikoysk/reposilite/blob/c38cee03571101016342568fe7550c97758825a8/reposilite-backend/src/test/kotlin/com/reposilite/maven/MavenIntegrationTest.kt#L73
